docker/test/base/setup_export_logs.sh

```diff
@@ -5,61 +5,150 @@
+                host: '${CLICKHOUSE_CI_LOGS_HOST}'
+                port: 9440
+                password: '${CLICKHOUSE_CI_LOGS_PASSWORD}'
+" > "$cluster_config"
+    echo "Cluster ${CLICKHOUSE_CI_LOGS_CLUSTER} is confugured in ${cluster_config}"
+)
+
+function setup_logs_replication
+(
+    # The function is launched in a separate shell instance to not expose the
+    # exported values from CLICKHOUSE_CI_LOGS_CREDENTIALS
+    set +x
+    # disable output
+    if ! [ -r "${CLICKHOUSE_CI_LOGS_CREDENTIALS}" ]; then
+        echo "File $CLICKHOUSE_CI_LOGS_CREDENTIALS does not exist, do not setup"
+        return 0
+    fi
+    set -a
+    # shellcheck disable=SC1090
+    source "${CLICKHOUSE_CI_LOGS_CREDENTIALS}"
+    set +a
+    # The only way to substitute the env as a plain text is using perl 's/\Qsomething\E/another/
+    exec &> >(perl -pe '
+        s(\Q$ENV{CLICKHOUSE_CI_LOGS_HOST}\E)[CLICKHOUSE_CI_LOGS_HOST]g;
+        s(\Q$ENV{CLICKHOUSE_CI_LOGS_USER}\E)[CLICKHOUSE_CI_LOGS_USER]g;
+        s(\Q$ENV{CLICKHOUSE_CI_LOGS_PASSWORD}\E)[CLICKHOUSE_CI_LOGS_PASSWORD]g;
+    ')
+    echo "Checking if the credentials work"
+    _check_logs_credentials || return 0
+    # It's impossible to use generous $CONNECTION_ARGS string, it's unsafe from word splitting perspective.
+    # That's why we must stick to the generated option
+    CONNECTION_ARGS=(--secure --user "${CLICKHOUSE_CI_LOGS_USER}" --host "${CLICKHOUSE_CI_LOGS_HOST}" --password "${CLICKHOUSE_CI_LOGS_PASSWORD}")
+
+    echo 'Create all configured system logs'
+    clickhouse-client --query "SYSTEM FLUSH LOGS"
+
+    # It's doesn't make sense to try creating tables if SYNC fails
+    echo "SYSTEM SYNC DATABASE REPLICA default" | clickhouse-client --receive_timeout 180 "${CONNECTION_ARGS[@]}" || return 0
+
+    # For each system log table:
+    echo 'Create %_log tables'
+    clickhouse-client --query "SHOW TABLES FROM system LIKE '%\\_log'" | while read -r table
+    do
+        # Calculate hash of its structure:
+        hash=$(clickhouse-client --query "
+            SELECT sipHash64(groupArray((name, type)))
+            FROM (SELECT name, type FROM system.columns
+                WHERE database = 'system' AND table = '$table'
+                ORDER BY position)
+            ")
+
+        # Create the destination table with adapted name and structure:
+        statement=$(clickhouse-client --format TSVRaw --query "SHOW CREATE TABLE system.${table}" | sed -r -e '
+            s/^\($/('"$EXTRA_COLUMNS"'/;
+            s/ORDER BY \(/ORDER BY ('"$EXTRA_ORDER_BY_COLUMNS"'/;
+            s/^CREATE TABLE system\.\w+_log$/CREATE TABLE IF NOT EXISTS '"$table"'_'"$hash"'/;
+            /^TTL /d
+            ')
+
+        echo -e "Creating remote destination table ${table}_${hash} with statement:\n${statement}" >&2
+
+        echo "$statement" | clickhouse-client --distributed_ddl_task_timeout=10 --receive_timeout=10 --send_timeout=10 "${CONNECTION_ARGS[@]}" || continue
```

Comment:
Let's also add `database_replicated_initial_query_timeout_sec=10` just in case

Reply:
maybe, we should add it globally to `CONNECTION_ARGS`?

Reply:
receive_timeout and send_timeout - maybe yes
distributed_ddl_task_timeout and database_replicated_initial_query_timeout_sec make sense only for these CREATE queries

Reply:
we have receive_timeout=180 in another place https://github.com/ClickHouse/ClickHouse/pull/53491/files#diff-26c882b0dca7da3859cf799ce30cbda5f0325e35e5ed8410e408c8d0a2036296R108